refactor: use CustomWebElement helpers in HomePage and CreateProfilePage

FlowerWineMonk opened the pull request:
- Wrapped interactions (select, dropdown, date, click) in CustomWebElement methods
  to handle waits and actions consistently
- Replaced direct waits in HomePage and CreateProfilePage with custom helpers
- Simplified test steps for cleaner, more reusable page methods

FlowerWineMonk committed:
refactor: use CustomWebElement helpers in HomePage and CreateProfilePage

- Wrapped interactions (select, dropdown, date, click) in
CustomWebElement methods
  to handle waits and actions consistently
- Replaced direct waits in HomePage and CreateProfilePage with custom
helpers
- Simplified test steps for cleaner, more reusable page methods

Anassstasi commented on src/main/java/vention/customWebElement/CustomWebElement.java:
```diff
@@ -44,6 +44,42 @@ public Select asSelect() {
     return new Select(getWrappedElement());
   }
 
+  public void customSelectFare(CustomWebElement fareId, String fareWay, int timeout) {
```
I'd recommend to remove 'fare' from naming as it narrows down the usage of this method. 

Basically it's just selectByValue and can be used for ANY value not just fare.

Anassstasi commented on src/main/java/vention/customWebElement/CustomWebElement.java:
```diff
@@ -44,6 +44,42 @@ public Select asSelect() {
+  }
+
+  public void customDropdownReload(CustomWebElement dropdownId, String dropdownKeys,
+      CustomWebElement dynamicAirportOption, String cityInfo, int timeout) {
```
Please fix code formatting

Anassstasi commented on src/main/java/vention/customWebElement/CustomWebElement.java:
```diff
@@ -44,6 +44,42 @@ public Select asSelect() {
+    reloadedOption.click();
+  }
+
+  public void customDepartureDate(CustomWebElement departureDateButton, CustomWebElement dynamicDateOption, String date,
```
Please update method name. It should start with a verb (in this case it should be 'click')

CustomWebElement - is a core class, its methods should be reusable for any entity/value, when you are creating methods like 'selectFare' or 'selectDepartureDate' it's impossible to reuse it for other entities (even though it can be)

This method is just a more powerful click, you might need it for other elements as well (the app you are testing spesifics are so), i'd recommend to create a method like 'waitForElementReloadAndClick()' - with not arguments as all the needed data you can get from the element object you are calling this method from

Anassstasi commented on src/main/java/vention/customWebElement/CustomWebElement.java:
```diff
@@ -44,6 +44,42 @@ public Select asSelect() {
+    reloadedDateOption.click();
+  }
+
+  public void customClick(CustomWebElement findFlightsButton, int timeout) {
```
Why do u need to pass CustomWebElement as an argument?

Anassstasi replied:
findFlightsButton is not a proper name for an argument in a generic method. What is someone passes not a button there? or a different button?

Anassstasi commented on src/main/java/vention/customWebElement/CustomWebElement.java:
```diff
@@ -44,6 +44,42 @@ public Select asSelect() {
+    reloadedFindFlightsButton.click();
+  }
+
+  public void customSelect(CustomWebElement element, String value, String cssSelector, int timeout) {
```
Same comments as above

Anassstasi reviewed (changes requested):
Left comments to fix

FlowerWineMonk committed:
refactor: rename and generalize CustomWebElement

- Replaced methods (customDepartureDate, customSelectFare, etc.)
  with generic, verb-based methods (waitForReloadAndClick,
clearAndTypeWithReload, etc.)
- Improved reusability of CustomWebElement for multiple entities
- Updated PageObject methods to use new core utilities

FlowerWineMonk committed:
chore: update gitignore and make application.properties visible

FlowerWineMonk committed:
refactor: change enum handling and chain WaitUtils methods calls directly to reduce memory usage
